include/llbuild/Basic/SignalHandler.h

```diff
@@ -0,0 +1,35 @@
+//===- SignalHandler.h ------------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements a wrapper for handling signals.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_BASIC_SIGNALHANDLER_H
+#define LLBUILD_BASIC_SIGNALHANDLER_H
+
+namespace llbuild {
+namespace basic {
+struct SignalHandler {
```

Comment:
We need some documentation comments here.

What I think we want the API to look like is something more like one method which registers an *interrupt* handler (ideally a lambda) specifically (I don't think we need to worry about any other signals at the moment, and other signals come with other problems).

That should return a token the client can use to unregister it.

That should probably be a static method, since there should only be one instance of the thing managing the handlers.

Internally, the manager for the handlers will simply keep a mapping between tokens and handlers (the token can be as simple as an ID we just increment, or an opaque pointer) and when an interrupt happens it will just call them all in some unspecified order.